### PR TITLE
`go.Figure.set_subplots`

### LINKED ISSUE #2405: fig.set_subplots
If we could call `set_subplots` on a pre-existing, sufficiently-empty figure, we could avoid a separate import just for `make_subplots` and have a smaller entry-point footprint .

### PR DESCRIPTION
**Pull request by nicholas-esterer**
Closes #2405 

This is now possible

```python
import plotly.graph_objects as go
figgy=go.Figure().set_subplots(2,3)
```
and `figgy` is equivalent to what is given by:

```python
from plotly.subplots import make_subplots
figgy=make_subplots(2,3)
```

## Code PR

- [ x] I have read through the [contributing notes](https://github.com/plotly/plotly.py/blob/master/contributing.md) and understand the structure of the package. In particular, if my PR modifies code of `plotly.graph_objects`, my modifications concern the `codegen` files and not generated files.
- [ x] I have added tests (if submitting a new feature or correcting a bug) or
  modified existing tests.
- [x] For a new feature, I have added documentation examples in an existing or
  new tutorial notebook (please see the doc checklist as well).
- [x] I have added a CHANGELOG entry if fixing/changing/adding anything substantial.

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
@jonmmease mind taking a quick look here to see if everything makes sense?

File: packages/python/plotly/plotly/tests/test_core/test_graph_objs/test_figure.py [hunk 1]
```diff
@@ -199,3 +200,25 @@ def test_update_overwrite_data(self):
...
+        # Test that it accepts the same arguments as make_subplots
+        fig1 = go.Figure()
+        fig1.set_subplots(rows=2, cols=2, horizontal_spacing=0.25, vertical_spacing=0.1)
+        fig1_sp = make_subplots(
+            rows=2, cols=2, horizontal_spacing=0.25, vertical_spacing=0.1
+        )
+        assert fig1.layout == fig1_sp.layout
+        # Test that calling on a figure that already has subplots throws an error.
+        raised = False
+        try:
```
Comment (jonmmease) [inline packages/python/plotly/plotly/tests/test_core/test_graph_objs/test_figure.py:219]: I'd recommend using `with pytest.raises` (https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions) here as it's a little more concise.

**Review by jonmmease — APPROVED** [review]
This looks like a nice addition!  One small suggestion on the tests, but otherwise I'm happy with it.

File: packages/python/plotly/plotly/tests/test_core/test_graph_objs/test_figure.py (hunk 1 above)
Comment (nicholas-esterer) [reply packages/python/plotly/plotly/tests/test_core/test_graph_objs/test_figure.py:219]: Originally I used `pytest.raisesRegex` but it broke the Python2 tests.

Comment (jonmmease) [reply packages/python/plotly/plotly/tests/test_core/test_graph_objs/test_figure.py:219]: Does the `match` argument to `pytest.raises` work with Python 2.7? e.g. `pytest.raises(ValueError, match=r".* 123 .*")`.

Comment (nicholas-esterer) [reply packages/python/plotly/plotly/tests/test_core/test_graph_objs/test_figure.py]: :+1:

**Pull request merged**